src/code/sys_cmpdma.c

```diff
@@ -19,14 +19,152 @@
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/sys_cmpdma/func_80178A94.s")
 
+typedef union {
+    struct {
+        u8 r;
+        u8 g;
+        u8 b;
+        u8 a;
+    };
+    s32 rgba;
+} Struct1_80178AC0;
+
+typedef union {
+    u16 a, b;
+    s32 ab;
+} Struct2_80178AC0;
+
+typedef struct {
+    /* 0x00 */ u32 data1;
+    /* 0x00 */ u32 data2;
+} CmpDmaBssStruct;
+
+// static CmpDmaBssStruct D_801FBBE0;
+extern CmpDmaBssStruct D_801FBBE0;
+
+#ifdef NON_MATCHING
+// Some constants re-ordered at the top
+void func_80178AC0(u16* src, void* dst, u32 size) {
+    Struct1_80178AC0 spC;
+    Struct2_80178AC0 sp8;
+    Struct2_80178AC0 sp4;
+    s32* phi_v0 = dst;
+    u16* phi_v1 = src;
+    u32 tempsp4;
+    u32 tempsp8;
+    s32* phi_a0;
+
+    if ((u32)((u8*)dst - (size & 0xFFFFFFFFFFFFFFFF)) >= (u32)dst) {
+        return;
+    }
+
+    phi_a0 = (s32*)((u8*)dst - size);
+
+    do {
+        sp8.a = *(phi_v1++);
+        phi_a0++;
+
+        if ((sp8.a & 1) == 1) {
+            tempsp8 = sp8.ab;
+
+            spC.r = ((tempsp8 >> 0x1B) * 0xFF) / 31;
+            spC.g = (((tempsp8 << 5) >> 0x1B) * 0xFF) / 31;
+            spC.b = (((tempsp8 << 0xA) >> 0x1B) * 0xFF) / 31;
+            spC.a = 0xFF;
+
+        } else if (sp8.a == 0) {
+            spC.rgba = 0;
+
+        } else {
+            sp4.a = sp8.a;
+            sp8.a = *(phi_v1++);
+
+            tempsp4 = sp4.ab;
+            tempsp8 = sp8.ab;
+
+            spC.r = ((tempsp8 >> 0x1B) * 8) | (tempsp4 >> 0x1D);
+            spC.g = (((tempsp8 << 5) >> 0x1B) * 8) | ((tempsp4 * 8) >> 0x1D);
+            spC.b = (((tempsp8 << 0xA) >> 0x1B) * 8) | ((tempsp4 << 6) >> 0x1D);
+            tempsp8 = sp8.a;
+            spC.a = ((tempsp8 & 1) << 7) | ((((tempsp4 << 9) >> 0x1B) * 0x7F) / 63);
+        }
+
+        *(phi_v0++) = spC.rgba;
+    } while ((u32)phi_a0 < (u32)dst);
+}
+#else
+void func_80178AC0(u16* src, void* dst, u32 size);
 #pragma GLOBAL_ASM("asm/non_matchings/code/sys_cmpdma/func_80178AC0.s")
+#endif
+
+void func_80178C80(u8* vrom, s32 id, s32* start, s32* size, s32* flag) {
```

Comment:
```suggestion
void func_80178C80(u8* rom, s32 id, s32* start, s32* size, s32* flag) {
```
Looks like it is the first arg to `DmaRomToRam` so this would be rom.